cmd/utils.go

```diff
@@ -168,104 +173,125 @@ func parseAddress(ctx context.Context, addr string, lapi lotusapi.FullNode) (com
 	return common.HexToAddress(ethAddr.String()), nil
 }
 
-func commonSetupOwnerCall() (common.Address, *ecdsa.PrivateKey, *ecdsa.PrivateKey, error) {
+func commonSetupOwnerCall() (agentAddr common.Address, ownerWallet accounts.Wallet, ownerAccount accounts.Account, ownerPassphrase string, requesterKey *ecdsa.PrivateKey, err error) {
 	as := util.AgentStore()
-	ks := util.KeyStore()
-	// Check if an agent already exists
-	agentAddrStr, err := as.Get("address")
-	if err != nil {
-		return common.Address{}, nil, nil, err
-	}
 
-	if agentAddrStr == "" {
-		return common.Address{}, nil, nil, errors.New("No agent found. Did you forget to create one?")
+	ownerAddr, _, err := as.GetAddrs(util.OwnerKey)
+	if err != nil {
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	agentAddr := common.HexToAddress(agentAddrStr)
-
-	pk, err := ks.GetPrivate(util.OwnerKey)
+	agentAddr, wallet, account, passphrase, requesterKey, err := commonOwnerOrOperatorSetup(context.Background(), ownerAddr.String())
 	if err != nil {
-		return common.Address{}, nil, nil, err
+		return common.Address{}, nil, accounts.Account{}, "", nil, err
 	}
 
-	if pk == nil {
-		return common.Address{}, nil, nil, errors.New("Owner key not found. Please check your `keys.toml` file.")
-	}
+	return agentAddr, wallet, account, passphrase, requesterKey, nil
+}
 
-	requesterKey, err := ks.GetPrivate(util.RequestKey)
+func commonOwnerOrOperatorSetup(ctx context.Context, from string) (agentAddr common.Address, wallet accounts.Wallet, account accounts.Account, passphrase string, requesterKey *ecdsa.PrivateKey, err error) {
```

Comment:
This should return the `auth` object itself (same comment as above)

Reply:
I've updated the common*Setup() calls to do that. Definitely cleaner.